code/datums/ability.dm

```diff
@@ -148,6 +155,11 @@
 			return
 		if(isnull(toggling))
 			toggling = !enabled
+		if(targeted)
+			if(ishuman(user) && toggling == TRUE)
+				var/mob/living/carbon/human/H = user
+				if(H.ab_handler)
+					H.ab_handler.process_ability(src)
```

Comment:
1. combine if statements if possible and sensical; it is sensical here
2. H.ab_handler?.process_ability(src)